wasm-test-app/src/Utils.tsx

```diff
@@ -0,0 +1,39 @@
+export function readUploadedFileAsText(file: File) {
+    return new Promise<Uint8ClampedArray>((resolve, reject) => {
+        const reader = new FileReader();
+
+        reader.onload = event => {
+            if (event.target && event.target.result) {
+                const arrayBuffer = event.target.result as ArrayBuffer;
+                resolve(new Uint8ClampedArray(arrayBuffer));
+            } else {
+                reject(new Error("Failed to read file"));
+            }
+        };
+
+        reader.onerror = error => {
+            reject(new Error('File could not be read: ' + error));
+        };
+
+        reader.readAsArrayBuffer(file);
+    });
+}
+// Custom hook for file download
+export const fileDownload = (fileName: string, buffer: ArrayBufferLike) => {
```

Comment:
I'm a bit confused on what's going on here. React hook NEED to start with `use`, custom of otherwise: https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use

Reply:
Yeah thats not a hook. Its a helper function, I will remove that comment for clarification.